src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -127,17 +150,13 @@
  (fn [{:keys [db]}]
    (let [web3-wallet          (get db :wallet-connect/web3-wallet)
          current-proposal     (get db :wallet-connect/current-proposal)
-         accounts             (get-in db [:wallet :accounts])
          supported-chain-ids  (->> db
                                    chain/chain-ids
                                    (map wallet-connect-core/chain-id->eip155)
                                    vec)
-         ;; NOTE: for now using the first account, but should be using the account selected by the
-         ;; user on the connection screen. The default would depend on where the connection started
-         ;; from:
-         ;; - global scanner -> first account in list
-         ;; - wallet account dapps -> account that is selected
-         address              (-> accounts keys first)
+         address              (-> db
+                                  (get :wallet-connect/current-proposal-account)
+                                  :address)
```

Comment:
A small comment, we extract once using `get` and once using the keyword; we could just use the keyword for the first case to have a consisten code style.